nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -661,6 +662,35 @@ impl BundleCollection<'_> {
                         );
                     }
                 }
+
+                // Collect all of the log files on this sled
```

Comment:
Is there a concern that we'll encounter a TOCTTOU here, if a log is rotated between the call to `support_logs` and the subsequent call to `save_log_file`?

If a log was *just* rotated, it's probably likely to be a relatively fresh/interesting logfile too, right?

Reply:
That is something that can surly happen.
When we first discussed this we debated sending all of the logs at once with some sort of delimiter in between but we decided that retrieving the logs one by one would the better route. I am not entirely sure how we would work around this with the current design.

Reply:
It may be worth looking at what `sled-agent/src/zone_bundle.rs` has done in this space - I recall our zone bundle logic encountering corruption due to concurrent log rotation. I believe the solution used there involved snapshotting the filesystem?

I'm not sure it *needs* to hold up this PR, but it's definitely worth fixing - this not only impacts the "most recent" log rotated, but could also be a problem for any "older" log being rotated from e.g. `log.1` to `log.2`.

Reply:
This was a really thorny problem in the zone-bundle case, and I spent a ton of time getting an uncorrupted copy of all the log files. I hit the issue both with rotation but also just the logs continually growing while we tried to bundle them up. The solution we ended up with creates a ZFS snapshot of the zone filesystem, copies all those logs, and then deletes the snapshot at the end. I don't think we've seen any issues since I implemented that part.

Reply:
Most of this feedback I can address in a follow up PR but this issue might require a bit more planning. Let's use today's support bundle sync to talk through this. Using a snapshot would solve this issue but it somehow has to be tied to the lifecycle of support bundle creation, for example since we currently fetch an index and then fetch each log file individually how long would the snapshot exist and do we take one on every index fetch etc.  We may need to somehow tie these api endpoints to a support bundle id so that we can create the relevant snapshots and also clean up the snapshot when a support bundle is removed.